src/ControlSystem/Systems/GridCenters.hpp

```diff
@@ -0,0 +1,133 @@
+      "GridCenters only accepts BothNSCenters measurement currently.");
+
+  using control_error = ControlErrors::GridCenters;
+  static_assert(tt::conforms_to_v<control_error,
+                                  control_system::protocols::ControlError>);
+
+  // tag goes in control component
+  struct MeasurementQueue : db::SimpleTag {
+    using type = LinkedMessageQueue<
+        double,
+        tmpl::list<
+            QueueTags::Center<::domain::ObjectLabel::A, Frame::Grid>,
+            QueueTags::Center<::domain::ObjectLabel::B, Frame::Grid>,
+            QueueTags::Center<::domain::ObjectLabel::A, Frame::Inertial>,
+            QueueTags::Center<::domain::ObjectLabel::B, Frame::Inertial>>>;
+  };
+
+  using simple_tags = tmpl::list<MeasurementQueue>;
+
+  struct process_measurement {
+    template <typename Submeasurement>
+    using argument_tags = tmpl::list<
+        measurements::Tags::NeutronStarCenter<::domain::ObjectLabel::A,
+                                              Frame::Grid>,
+        measurements::Tags::NeutronStarCenter<::domain::ObjectLabel::B,
+                                              Frame::Grid>,
+        measurements::Tags::NeutronStarCenter<::domain::ObjectLabel::A,
+                                              Frame::Inertial>,
+        measurements::Tags::NeutronStarCenter<::domain::ObjectLabel::B,
+                                              Frame::Inertial>>;
+
+    template <typename Metavariables>
+    static void apply(
+        measurements::BothNSCenters::FindTwoCenters submeasurement,
+        const std::array<double, 3> grid_center_a,
+        const std::array<double, 3> grid_center_b,
+        const std::array<double, 3> inertial_center_a,
+        const std::array<double, 3> inertial_center_b,
+        Parallel::GlobalCache<Metavariables>& cache,
+        const LinkedMessageId<double>& measurement_id) {
+      auto& control_sys_proxy = Parallel::get_parallel_component<
+          ControlComponent<Metavariables, GridCenters>>(cache);
+
+      Parallel::simple_action<::Actions::UpdateMessageQueue<
+          MeasurementQueue, UpdateControlSystem<GridCenters>,
+          QueueTags::Center<::domain::ObjectLabel::A, Frame::Grid>,
+          QueueTags::Center<::domain::ObjectLabel::B, Frame::Grid>,
+          QueueTags::Center<::domain::ObjectLabel::A, Frame::Inertial>,
+          QueueTags::Center<::domain::ObjectLabel::B, Frame::Inertial>>>(
+          control_sys_proxy, measurement_id, DataVector(grid_center_a),
+          DataVector(grid_center_b), DataVector(inertial_center_a),
+          DataVector(inertial_center_b));
+
+      if (Parallel::get<Tags::Verbosity>(cache) >= ::Verbosity::Verbose) {
+        Parallel::printf("%s, time = %.16f: Received measurement '%s'.\n",
+                         name(), measurement_id.id,
+                         pretty_type::name(submeasurement));
+      }
+    }
+  };
+};
+}  // namespace control_system::Systems
```

Comment:
[random line] Please also add tests of the control error and system. They can both be simple tests like `Test_Skew.cpp` in their respective directories.